qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -387,7 +385,30 @@ def simplify(self, atol=None, rtol=None):
 
         # Pack bool vectors into np.uint8 vectors by np.packbits
         array = np.packbits(self.paulis.x, axis=1) * 256 + np.packbits(self.paulis.z, axis=1)
-        _, indexes, inverses = np.unique(array, return_index=True, return_inverse=True, axis=0)
+
+        def _unique(array):
+            # This function corresponds to
+            # _, indexes, inverses = np.unique(array, return_index=True, return_inverse=True, axis=0)
+            #
+            # But, note that this does not sort the array while np.unique sorts the array.
+            table = {}
+            indexes = []
+            inverses = np.empty(array.shape[0], dtype=int)
+            for i, ary in enumerate(array):
+                b = ary.data.tobytes()
+                if b in table:
+                    inverses[i] = table[b]
+                else:
+                    inverses[i] = table[b] = len(table)
+                    indexes.append(i)
+            return indexes, inverses
```

Comment:
This will have better asymptotic scaling (`O(n_ops * n_qubits)` as opposed to `O(n_ops * ln(n_ops) * n_qubits)`) but it might not be the best in all situations - the Python loop iteration will have much larger constant factors than Numpy's looping.  If this is a core inner-loop function, it might be worth having two algorithms for it.